src/_pytest/unittest.py

```diff
@@ -1,27 +1,37 @@
 """ discovery and running of std-library "unittest" style tests. """
 import sys
 import traceback
+from typing import Iterable
+from typing import Optional
+from typing import Union
 
 import _pytest._code
 import pytest
 from _pytest.compat import getimfunc
 from _pytest.config import hookimpl
+from _pytest.nodes import Collector
+from _pytest.nodes import Item
 from _pytest.outcomes import exit
 from _pytest.outcomes import fail
 from _pytest.outcomes import skip
 from _pytest.outcomes import xfail
 from _pytest.python import Class
 from _pytest.python import Function
+from _pytest.python import PyCollector
 from _pytest.runner import CallInfo
 
 
-def pytest_pycollect_makeitem(collector, name, obj):
+def pytest_pycollect_makeitem(
+    collector: PyCollector, name: str, obj
+) -> Optional["UnitTestCase"]:
     # has unittest been imported and is obj a subclass of its TestCase?
     try:
-        if not issubclass(obj, sys.modules["unittest"].TestCase):
-            return
+        ut = sys.modules["unittest"]
+        # Type ignored because ut is an opaque module.
+        if not issubclass(obj, ut.TestCase):  # type: ignore
```

Comment:
I'd still prefer to see the ignore codes here.

Reply:
I also think it's a good idea to add them. But let's maybe first add `show-error-codes` to the mypy config to make it "official" and show up by default in editor/CI/etc. WDYT?

Reply:
> But let's maybe first add show-error-codes to the mypy config to make it "official" and show up by default in editor/CI/etc.

:+1: 
(FWIW I have it in my editor's config already, and I think we can use them without it in the default config already.  Anyway, should not hold this back in general, but the next time maybe.. :))